src/sys/event.rs

```diff
@@ -293,30 +454,19 @@ type type_of_nchanges = c_int;
 #[cfg(target_os = "netbsd")]
 type type_of_nchanges = size_t;
 
-pub fn kevent_ts<Fd: AsFd>(
-    kq: Fd,
+#[allow(missing_docs)]
+#[deprecated(since = "0.27.0", note = "Use Kqueue::kevent instead")]
+pub fn kevent_ts(
+    kq: &Kqueue,
     changelist: &[KEvent],
     eventlist: &mut [KEvent],
     timeout_opt: Option<timespec>,
 ) -> Result<usize> {
-    let res = unsafe {
-        libc::kevent(
-            kq.as_fd().as_raw_fd(),
-            changelist.as_ptr() as *const libc::kevent,
-            changelist.len() as type_of_nchanges,
-            eventlist.as_mut_ptr() as *mut libc::kevent,
-            eventlist.len() as type_of_nchanges,
-            if let Some(ref timeout) = timeout_opt {
-                timeout as *const timespec
-            } else {
-                ptr::null()
-            },
-        )
-    };
-
-    Errno::result(res).map(|r| r as usize)
+    kq.kevent(changelist, eventlist, timeout_opt)
 }
 
+/// Modify an existing [`KEvent`].
+// Probably should deprecate.  Would anybody ever use it over `KEvent::new`?
```

Comment:
I think deprecating it makes sense. If there is a use case I expect we'll hear about it after marking it as deprecated - we can always remove the deprecation label and keep it around if a use case does come up.